### PR TITLE
CORE-13380: Label Image With Source Information

### PR DESCRIPTION
**Pull request by jujoramos**
Add the following labels to the built images:
  - "com.r3.corda.git.branch": branch used.
  - "org.opencontainers.image.source": remote git url.
  - "org.opencontainers.image.revision": git commit hash.

### CODE REVIEW

**Comment by ronanbrowne** [comment]
> Rule: 26-GENERIC-PASSWORD Severity: CRITICALFile: .run/db-worker-local-debug.sh Lines: 10 Scanner: [Secret Diver](https://github.com/cider-rnd/secret-diver)
> 
> More Info
> Rule: 2-PEM-PRIVATE Severity: CRITICALFile: applications/workers/release/combined-worker/tls/rest/server.key Lines: 1 Scanner: [Secret Diver](https://github.com/cider-rnd/secret-diver)
> 
> More Info
> Rule: 2-PEM-PRIVATE Severity: CRITICALFile: docs/kafka/security/demo/certificates/ca/ca-key Lines: 1 Scanner: [Secret Diver](https://github.com/cider-rnd/secret-diver)
> 
> More Info
> Rule: 2-PEM-PRIVATE Severity: CRITICALFile: libs/rest/ssl-cert-read-impl/src/test/resources/server.key Lines: 1 Scanner: [Secret Diver](https://github.com/cider-rnd/secret-diver)
> 
> More Info

Config issue not related to this PR can ignore. This has been resolved elsewhere.

**Comment by jujoramos** [comment]
Below is the output of inspecting the labels for the `flow-worker` image generated by Jenkins for this `PR`:

```
$$> CHART_VERSION=5.1.0-alpha.1683620017673

$$> APP_VERSION=$(helm show chart "oci://corda-os-docker-dev.software.r3.com/helm-charts/pr-3825/corda" --version "${CHART_VERSION}" | grep appVersion | awk '{print $2}')
Pulled: corda-os-docker-dev.software.r3.com/helm-charts/pr-3825/corda:5.1.0-alpha.1683620017673
Digest: sha256:14a984b8f3bdc6e1d61d20c43e4e58613b23d18edce7069bbb77cc5c3903e53f

$$> echo "${APP_VERSION}"
5.1.0.0-alpha-1683620017673

$$> docker pull "corda-os-docker.software.r3.com/corda-os-flow-worker:${APP_VERSION}"
5.1.0.0-alpha-1683620017673: Pulling from corda-os-flow-worker
455c32d33260: Pull complete 
b8031bf770c0: Pull complete 
263b053707ff: Pull complete 
Digest: sha256:35f3b1d25aa56fa012a9bbb107650d2390e34a7b14f704a227e5da99712f8c53
Status: Downloaded newer image for corda-os-docker.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683620017673
corda-os-docker.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683620017673

$$> docker image inspect "corda-os-docker.software.r3.com/corda-os-flow-worker:${APP_VERSION}" | grep -A 5 -F "\"Labels\": {" | grep -E "branch|revision|source"
                "com.r3.corda.git.branch": "HEAD",
                "org.opencontainers.image.revision": "3b473bc6172d5a5c7052dd233f8787abcab55e70",
                "org.opencontainers.image.source": "https://github.com/corda/corda-runtime-os.git",
```

File: buildSrc/src/main/groovy/DeployableContainerBuilder.groovy [hunk 1]
```diff
@@ -177,8 +180,21 @@ abstract class DeployableContainerBuilder extends DefaultTask {
...
+        gitBranchTask = project.tasks.register("gitBranch", GetGitBranch.class)
+        super.dependsOn(gitBranchTask)
+
+        gitRemoteTask = project.tasks.register("gitRemote", GetGitRemoteUrl.class)
+        super.dependsOn(gitRemoteTask)
+
+        gitRevisionTask = project.tasks.register("gitRevision", GetGitRevision.class)
+        super.dependsOn(gitRevisionTask)
+
+        // TODO: remove once pipelines have been updated to consume images using the full hash as the tag.
```
Comment (ronanbrowne) [inline buildSrc/src/main/groovy/DeployableContainerBuilder.groovy]: Is there a JIra for this ?

**Review by ronanbrowne — DISMISSED** [review]
LGTM

File: buildSrc/src/main/groovy/DeployableContainerBuilder.groovy (hunk 1 above)
Comment (jujoramos) [reply buildSrc/src/main/groovy/DeployableContainerBuilder.groovy]: Not at the moment, I'll create one and update the `TODO` with the actual link. Thanks for flagging this 👍 .

**Review by ronanbrowne — DISMISSED** [review]

**Review by ronanbrowne — APPROVED** [review]

**Comment by jujoramos** [comment]
Labels on latest image built from this `PR`:

```
$$> docker pull "corda-os-docker-dev.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683648900380"
5.1.0.0-alpha-1683648900380: Pulling from corda-os-flow-worker
455c32d33260: Already exists 
b8031bf770c0: Already exists 
ddc190e65fa4: Pull complete 
Digest: sha256:9589e1d9dc4a9a0a724beade228872484a5967ad5f54462d2fe0e9e7d01966a9
Status: Downloaded newer image for corda-os-docker-dev.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683648900380
corda-os-docker-dev.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683648900380



$$> docker inspect "corda-os-docker-dev.software.r3.com/corda-os-flow-worker:5.1.0.0-alpha-1683648900380" | grep -A 5 -F "\"Labels\": {"
            "Labels": {
                "com.r3.corda.git.branch": "jramos/CORE-13380",
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.revision": "0ce7c7ce6a634c698b546d85f540fa77ff11b00c",
                "org.opencontainers.image.source": "https://github.com/corda/corda-runtime-os.git",
                "org.opencontainers.image.version": "20.04"
```

**Pull request merged**